SharpRepository.EfRepository/EfRepositoryBase.cs

```diff
@@ -14,11 +14,12 @@ public class EfRepositoryBase<T, TKey> : LinqRepositoryBase<T, TKey> where T : c
     {
         protected IDbSet<T> DbSet { get; private set; }
         protected DbContext Context { get; private set; }
+        public bool GenerateKeyOnAdd { get; set; }
```

Comment:
I've pushed a new commit where I put it in ```ICrudRepository``` and implemented it in ```RepositoryBase``` and ```ConfigurationBasedRepository```. Added support in all repositories.

I'll check if it's possible to write a test for it in ```RepositoryAddTests```.